### ISSUE TITLE
Log Viewer Window delays redraw

### ISSUE THREAD

**Issue by mmikeww** [talonvoice/talon#648]
Windows10

1. keep Log Window open, but not focused/active
2. trigger log update, either built in, or with `print()`
3. log window will update, but the last line will be visually cut off
4. move mouse over Log Window and then the line is fully displayed

seems to be a graphics issue with whatever gui framework you're using, where it wont fully redraw or update its window contents until a mouse message is sent to the window

**Comment by lunixbochs** [comment]
which exact talon version? which OS?

**Comment by mmikeww** [comment]
talon 0.4.0 public (non beta)
win10 professional

low priority really. but i'm noticing the final line being cut off with my own personal print() debug statements

i thought it was happening with builtin log msgs too, but what i'm seeing now is, not necessarily that the final line gets cut off, but the whole window text just doesn't update until i mouse over, and then i see timestamps update etc. as if, similarly, the window needs to redraw and update its contents

**Comment by lunixbochs** [comment]
this is likely already fixed in the beta

**Comment by lunixbochs** [comment]
are you sure it's not printing? another option may be that it printed the line but didn't autoscroll quite far enough until you interacted.

**Comment by mmikeww** [comment]
yes exactly. its definitely printing. the autoscroll doesn't update until i mouseover the window. i know the line is printed because its cut off. i can visually see part of it.

i haven't signed up for the beta yet to test if its fixed in that or not

**Comment by mmikeww** [comment]
here.. first you can see some failed mousemoves trying to use zoom mouse inside a VM where the mouse wont move. then at the very end, i edited and saved a userscript and you can barely see the DEBUG line notifying of the change. but it doesn't come into view until i mouseover the window

![image](https://github.com/talonvoice/talon/assets/1900684/4c5638fa-493e-4359-9e4d-3b55f2694122)

**Comment by lunixbochs** [comment]
fixed in next beta
